File: C++/섹션06. DFS - 경우의 수 만들기/02-01. 중복 순열 (permutations with repetition), BOJ 15651 - N과 M (3).cpp
#include <stdio.h>

#define MAX (7 + 3)

int N, M;

int num_of_cases[MAX];

void printCases()
{
	for (int i = 0; i < M; i++) printf("%d ", num_of_cases[i]);
	putchar('\n');
}

void DFS(int depth)
{
	if (depth == M)
	{
		printCases();
		return;
	}

	for (int i = 1; i <= N; i++)
	{
		num_of_cases[depth] = i;

		DFS(depth + 1);		
	}
}

int main()
{
	scanf("%d %d", &N, &M);

	DFS(0);

	return 0;
}

File: Python/섹션06. DFS - 경우의 수 만들기/02-01. 중복 순열 (permutations with repetition), BOJ 15651 - N과 M (3).py
MAX = 7 + 3

N, M = map(int, input().split())

num_of_cases = [0] * MAX

def print_cases():
    print(*num_of_cases[:M])

def DFS(depth):
    if depth == M:
        print_cases()
        return
    
    for i in range(1, N + 1):
        num_of_cases[depth] = i
        
        DFS(depth + 1)
        
DFS(0)